### ISSUE TITLE
Minor Subtitle Issues

### ISSUE THREAD

**Issue by stdedos** [pytube/pytube#289]
Minor issue:
1. Cannot download "automatically generated" English subtitles (same key as "normal ones")

"Kind of" more important:
1. Caption 'lang' is in Finnish (I access YouTube from Finland)

Replicate:
On a video that has captions, from an IP geolocated "not in English-speaking countries" (and localized)

```python
[...]
pprint(yt.captions.all())
[...]
```

I see

```
[<Caption lang="Arabia" code="ar">,
 <Caption lang="Englanti (automaattisesti luotu)" code="en">,
 <Caption lang="Englanti" code="en">,
 <Caption lang="Espanja" code="es">,
 <Caption lang="Espanja (Espanja)" code="es-ES">,
 <Caption lang="Hollanti" code="nl">,
 <Caption lang="Japani" code="ja">,
 <Caption lang="Kiina (Taiwan)" code="zh-TW">,
 <Caption lang="Makedonia" code="mk">,
 <Caption lang="Puola" code="pl">,
 <Caption lang="Ranska" code="fr">,
 <Caption lang="Romania" code="ro">,
 <Caption lang="Saksa" code="de">,
 <Caption lang="Slovakki" code="sk">,
 <Caption lang="Suomi" code="fi">,
 <Caption lang="und" code="und">,
 <Caption lang="Ukraina" code="uk">,
 <Caption lang="Unkari" code="hu">]
```

**Comment by Mattwmaster58** [comment]
`<Caption lang="Englanti" code="en">`. Is that not English?

**Comment by stdedos** [comment]
Yes it is. "But so are these:" `<Caption lang="Englanti (automaattisesti luotu)" code="en">,`

**Comment by Mattwmaster58** [comment]
Ahh, so you can't download ` <Caption lang="Englanti (automaattisesti luotu)" code="en">`. Can you select that option in a browser?

**Comment by stdedos** [comment]
Yes, its the auto-generated English subtitles that exist in some videos.

**Comment by Mattwmaster58** [comment]
So does calling `captions[1].xml_captions` return `None`?

**Comment by stdedos** [comment]
Oops! I didn't think of that :/ My bad

Yes, it does 

```python
>>> with open('desktop/test.xml', 'w') as f:
...     f.write(yt.captions.all()[1].xml_captions)
...
69637

>>> with open('desktop/test.xml', 'w') as f:
...     f.write(yt.captions.all()[2].xml_captions)
...
33621
```

----

_For automatic captions, it's possible that you'd like to remove all:_ `<font color="#E5E5E5"></font>, <font color="#CCCCCC"></font>` (since automatic subtitles gradually appear in gray and later in time they become white as the speaker continues)

Example

```
2
000:000:10,160 --> 000:000:18,920
bigger welcome<font color="#E5E5E5"> hello San Francisco</font><font color="#CCCCCC"> TEDx</font>

3
000:000:15,620 --> 000:000:23,900
<font color="#CCCCCC">oh</font><font color="#E5E5E5"> my god blinding light hi everybody</font>
```

----

Feel free to handle (or not) the ticket as you which. 😄

**Comment by Mattwmaster58** [comment]
This issue is annoying cause I can't reproduce of course! What's the value of the `Caption`'s attribute `url`? Might as well tell me the non auto generated English subtitle `url` to.

**Comment by stdedos** [comment]
tl;dr: auto-generated have the `kind=asr` key

Auto-generated: https://www.youtube.com/api/timedtext?sparams=asr_langs%2Ccaps%2Cv%2Cxorp%2Cexpire&expire=1530420386&caps=asr&hl=fi_FI&signature=BF5729C2A51E334A03B65A32240A2C945B564D0D.51EB8F6F32C3A694F243F1FA21D6EBF40106030E&key=yttt1&asr_langs=es%2Cru%2Cnl%2Cpt%2Cde%2Cko%2Cit%2Cen%2Cja%2Cfr&v=some-id&xorp=True&kind=asr&lang=en

Engish: https://www.youtube.com/api/timedtext?sparams=asr_langs%2Ccaps%2Cv%2Cxorp%2Cexpire&expire=1530420386&caps=asr&hl=fi_FI&signature=BF5729C2A51E334A03B65A32240A2C945B564D0D.51EB8F6F32C3A694F243F1FA21D6EBF40106030E&key=yttt1&asr_langs=es%2Cru%2Cnl%2Cpt%2Cde%2Cko%2Cit%2Cen%2Cja%2Cfr&v=some-id&xorp=True&lang=en

**Comment by WebInMe** [comment]
Is it possible to add functionality for getting auto generated subtitles for language, that is not presented in Captions?
